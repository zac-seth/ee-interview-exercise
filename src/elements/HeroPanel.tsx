import { FC } from 'react';

import { Panel, PanelVariant } from './Panel.style';

export type HeroPanelProps = {
  children: JSX.Element;
};

export const HeroPanel: FC<HeroPanelProps> = ({ children }) => (
  <Panel variant={PanelVariant.HERO} data-testid='hero-panel'>
    {children}
  </Panel>
);
