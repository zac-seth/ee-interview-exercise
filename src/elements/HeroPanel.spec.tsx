import { render, screen } from '@testing-library/react';

import { HeroPanel } from './HeroPanel';

describe(`HeroPanel`, () => {
  it(`Should render a hero panel`, () => {
    render(
      <HeroPanel>
        <div>Hero content in a panel!</div>
      </HeroPanel>
    );

    expect(screen.getByTestId(`hero-panel`)).toMatchSnapshot();
  });

  it(`Should choose the hero panel background color style`, () => {
    render(
      <HeroPanel>
        <div>Hero content in a panel!</div>
      </HeroPanel>
    );

    const heroPanel = screen.getByTestId(`hero-panel`);

    expect(heroPanel).toHaveStyleRule(`background-color`, `#171717`);
  });
});
