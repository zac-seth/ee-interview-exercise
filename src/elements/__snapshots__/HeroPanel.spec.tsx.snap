// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroPanel Should render a hero panel 1`] = `
.c0 {
  width: 33.25rem;
  padding: .25rem;
  border-radius: .25rem;
  background-color: #FFF;
  background-color: #171717;
}

<section
  class="c0"
  data-testid="hero-panel"
>
  <div>
    Hero content in a panel!
  </div>
</section>
`;
